api/grpcserver/http_server.go

```diff
@@ -90,19 +89,21 @@ func (s *JSONHTTPServer) StartService(
 			err = pb.RegisterTransactionServiceHandlerServer(ctx, mux, typed)
 		case *DebugService:
 			err = pb.RegisterDebugServiceHandlerServer(ctx, mux, typed)
+		default:
+			s.logger.Error("unsupported service", zap.Stringer("service", svc))
 		}
 		if err != nil {
-			s.logger.Error("registering %T with grpc gateway failed with %v", svc, err)
+			s.logger.Error("registering with grpc gateway failed", zap.Stringer("service", svc), zap.Error(err))
 			return err
 		}
 		serviceCount++
 	}
 
-	s.logger.With().Info("starting grpc gateway server", log.String("address", s.listener))
+	s.logger.Info("starting grpc gateway server", zap.String("address", s.listener))
 
 	lis, err := net.Listen("tcp", s.listener)
 	if err != nil {
-		s.logger.Error("error listening: %v", err)
+		s.logger.Error("start listen server", zap.Error(err))
 		return err
```

Comment:
How about:
```suggestion
		return fmt.Errorf("listening on %s: %w", s.listener, err)
```

and let the caller do the logging. Similarly in other places. Reasoning: IMHO, the error should be logged in the place where it is handled, otherwise there is a high chance the same error will be logged multiple times. Bubble-ups should only add useful context for identifying the issue.

Reply:
While I do agree the logging message here is the only thing that happens when the JSON server fails to start, the caller (`node.go`) ignores any error that is returned from this method.

I changed to instead log in `node.go`, again not adding a panic, since going from completely ignoring a failing configuration to the node failing to start could break setups that might already be in place.

Reply:
>going from completely ignoring a failing configuration to the node failing to start could break setups that might already be in place.

These setups are already broken but don't know it :D.

Reply:
With my latest commit all services have been updated to be usable with the JSONServer and a failing start now results in the node panicing and shutting down.